sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2635,7 +2643,11 @@ object Sequence {
         var i = 0
 
         while (t < exclusiveItem ^ stepSign < 0) {
-          arr(i) = fromLong(t / scale)
+          arr(i) = if (scale == 1) {
+            fromLong(t / scale)
+          } else {
+            fromLong(Math.round(t / scale.toFloat))
```

Comment:
We could get date from `Math.round(t / scale.toFloat)`, if the target is timestamp,  the former ` fromLong(t / scale)` could be used.